controllers/torrent/download.go

```diff
@@ -7,27 +7,41 @@ import (
 	"strconv"
 	"strings"
 
+	msg "github.com/NyaaPantsu/nyaa/utils/messages"
+	"github.com/NyaaPantsu/nyaa/utils/upload"
+
 	"github.com/NyaaPantsu/nyaa/config"
-	"github.com/NyaaPantsu/nyaa/templates"
 	"github.com/NyaaPantsu/nyaa/models/torrents"
+	"github.com/NyaaPantsu/nyaa/templates"
 	"github.com/NyaaPantsu/nyaa/utils/format"
 	"github.com/gin-gonic/gin"
 )
 
 // DownloadTorrent : Controller for downloading a torrent
 func DownloadTorrent(c *gin.Context) {
 	hash := c.Param("hash")
+	messages := msg.GetMessages(c)
+
+	if hash == "" { // if no hash provided, you can't find a torrent in db neither the torrent file
```

Comment:
There is no /download/ route, only /download/:hash
If hash is empty then download.go will never get executed in the first place

Reply:
You're right but I don't know if we should really not check it.

Reply:
checked it